lib/ansible/modules/cloud/amazon/waf_facts.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+module: waf_facts
+short_description: Retrieve facts for WAF ACLs, Rule , Conditions and Filters.
+description:
+  - Retrieve facts for WAF ACLs, Rule , Conditions and Filters.
```

Comment:
I'm not really familiar with WAF yet so I might just be expecting the wrong thing from this module - it's only returning Global (Cloudfront) resources. Is that right? It ignores the ALB WAF resources in specific regions.

Along with renaming this module to begin with aws_, it maybe would also be clearer to call it aws_waf_acl_facts since it doesn't return anything not associated with a Web ACL?